src/transformers/models/patchtst/modeling_patchtst.py

```diff
@@ -0,0 +1,1958 @@
+    """
+
+    def __init__(
+        self,
+        sequence_length: int,
+        patch_length: int,
+        stride: int,
+        padding: bool = False,  # TODO: use this to set whether we want to pad zeros to the sequence
+    ):
+        super().__init__()
+
+        assert (
+            sequence_length > patch_length
+        ), f"Sequence length ({sequence_length}) has to be greater than the patch length ({patch_length})"
+
+        self.sequence_length = sequence_length
+        self.patch_length = patch_length
+        self.stride = stride
+
+        # get the number of patches
+        self.num_patches = compute_num_patches(sequence_length, patch_length, stride)
+        new_sequence_length = patch_length + stride * (self.num_patches - 1)
+        self.s_begin = sequence_length - new_sequence_length
+
+    def forward(self, past_values: torch.Tensor):
+        """
+        Parameters:
+            past_values (torch.Tensor, required): Input of shape [bs x sequence_length x num_input_channels]
+
+        Returns:
+            x: output tensor data [bs x num_input_channels x num_patches x patch_length]
+        """
+        sequence_length = past_values.shape[-2]
+        assert (
+            sequence_length == self.sequence_length
+        ), f"Input sequence length ({sequence_length}) doesn't match model configuration ({self.sequence_length})."
+
+        x = past_values[:, self.s_begin :, :]  # x: [bs x new_sequence_length x nvars]
+        x = x.unfold(
+            dimension=-2, size=self.patch_length, step=self.stride
+        )  # x: [bs x num_patches x num_input_channels x patch_length]
+        x = x.transpose(-2, -3).contiguous()  # xb: [bs x num_input_channels x num_patches x patch_length]
+        return x
+
+
+class PatchEmbeddings(nn.Module):
+    """
+    Parameters:
+    A class to patchify the time series sequence into different patches
+        sequence_length (int, required): input sequence length. patch_length (int, required): patch length. stride
+        (int, required): stride between patches.
+
+    Returns:
+        embeddings: output tensor data [bs x num_input_channels x num_patches x embed_dim]
+    """
+
+    def __init__(self, sequence_length: int, patch_length: int, stride: int, embed_dim: int):
+        super().__init__()
+
+        assert (
+            sequence_length > patch_length
+        ), f"Sequence length ({sequence_length}) has to be greater than the patch length ({patch_length})"
+
+        # assert ((max(sequence_length, patch_length) - patch_length) % stride == 0), f"sequence length minus patch length has to be divisible to the stride"
+
+        self.sequence_length = sequence_length
+        self.patch_length = patch_length
+        self.stride = stride
+        self.embed_dim = embed_dim
+
+        # get the number of patches
+        self.num_patches = compute_num_patches(sequence_length, patch_length, stride)
```

Comment:
this can be inferred from the config which already computes the number of patches

Reply:
Edit: better to do this in the model rather than the config (so could be deleted in the config)